source/common/http/utility.cc

```diff
@@ -380,6 +382,94 @@ Utility::getLastAddressFromXFF(const Http::HeaderMap& request_headers, uint32_t
   }
 }
 
+void Utility::sanitizeConnectionHeader(spdlog::logger& logger, Http::HeaderMap& headers) {
+  // Remove any headers nominated by the Connection header. The TE header
+  // is sanitized and removed only if it's empty after removing unsupported values
+  // See https://github.com/envoyproxy/envoy/issues/8623
+  const auto& cv = Http::Headers::get().ConnectionValues;
+  const auto& connection_header_value = headers.Connection()->value();
+
+  StringUtil::CaseUnorderedSet headers_to_remove{};
+
+  // Split the connection header and evaluate each nominated header
+  for (const auto& token : absl::StrSplit(connection_header_value.getStringView(), ',')) {
```

Comment:
I'm forgetting the type of token here.

If a malicious client sent a connection: gzip, connection, foo header can get us to delete of a header we're currently walking through?

Reply:
Good point!  `token` is a `std::string`.  Makes sense to check that we don't remove the Connection header we're operating on.   I don't think that there's a risk of concurrent modification here though.

Reply:
cool, sounds like we're safe.  do you mind adding a few unit tests that muck about with this potentially dangerous case, just so our sanitizer builds ensure we never refactor it in a dangerous way?

Reply:
Sure thing!